timestamp/request.go

```diff
@@ -85,15 +88,34 @@ func (r *Request) UnmarshalBinary(data []byte) error {
 	return err
 }
 
-// getOIDFromDigestAlgorithm returns corresponding ASN.1 OID for the given digest algorithm.
-func getOIDFromDigestAlgorithm(alg digest.Algorithm) (asn1.ObjectIdentifier, error) {
+// getOID returns corresponding ASN.1 OID for the given Hash algorithm.
+func getOID(alg crypto.Hash) (asn1.ObjectIdentifier, error) {
 	switch alg {
-	case digest.SHA256:
+	case crypto.SHA256:
 		return oid.SHA256, nil
-	case digest.SHA384:
+	case crypto.SHA384:
 		return oid.SHA384, nil
-	case digest.SHA512:
+	case crypto.SHA512:
 		return oid.SHA512, nil
 	}
-	return nil, digest.ErrDigestUnsupported
+	return nil, MalformedRequestError{msg: fmt.Sprintf("unsupported hashing algorithm: %s", alg)}
 }
+
+func validate(digest []byte, alg crypto.Hash) error {
+	l := len(digest)
+	var validContent bool
+	switch alg {
+	case crypto.SHA256:
+		validContent = l == 256/8
+	case crypto.SHA384:
+		validContent =  l == 384/8
+	case crypto.SHA512:
+		validContent = l == 384/8
```

Comment:
We are missing unit tests.

```suggestion
	case crypto.SHA512:
		validContent = l == 512/8
```